agent/internal/mapper/grpc.go

```diff
@@ -70,42 +71,180 @@ func MapDeployImage(req *agent.DeployRequest, appConfig *config.CommonConfigurat
 	return res
 }
 
-func MapContainerConfig(in *agent.DeployRequest_ContainerConfig) v1.ContainerConfig {
+func MapContainerConfig(in *agent.DeployRequest) v1.ContainerConfig {
 	containerConfig := v1.ContainerConfig{
-		Container: in.Name,
-		Ports:     MapPorts(in.Ports),
-		User:      &in.User,
+		Container:        in.Name,
+		ContainerPreName: in.InstanceConfig.Prefix,
+		Ports:            MapPorts(in.ContainerConfig.Ports),
+		PortRanges:       MapPortRanges(in.ContainerConfig.PortRanges),
+		Volumes:          MapVolumes(in.ContainerConfig.Volumes),
+		User:             in.ContainerConfig.User,
 	}
 
-	if in.Prefix != nil {
-		containerConfig.ContainerPreName = *in.Prefix
+	if in.ContainerConfig.TTY != nil {
+		containerConfig.TTY = *in.ContainerConfig.TTY
 	}
 
-	if in.Mounts != nil {
-		containerConfig.Mounts = in.Mounts
+	if in.ContainerConfig.Args != nil {
+		containerConfig.Args = in.ContainerConfig.Args
 	}
 
-	if in.Environments != nil {
-		containerConfig.Environment = in.Environments
+	if in.ContainerConfig.Command != nil {
+		containerConfig.Command = in.ContainerConfig.Command
 	}
 
-	if in.Expose != nil {
-		containerConfig.Expose = in.Expose.Public
-		containerConfig.ExposeTLS = in.Expose.Tls
+	if in.ContainerConfig.Expose != nil {
+		containerConfig.Expose = in.ContainerConfig.Expose.Public
+		containerConfig.ExposeTLS = in.ContainerConfig.Expose.Tls
 	}
 
-	if in.ConfigContainer != nil {
-		containerConfig.ConfigContainer = MapConfigContainer(in.ConfigContainer)
+	if in.ContainerConfig.Ingress != nil {
+		containerConfig.IngressName = in.ContainerConfig.Ingress.Name
+		containerConfig.IngressHost = in.ContainerConfig.Ingress.Host
+		containerConfig.IngressUploadLimit = *in.ContainerConfig.Ingress.UploadLimit
 	}
 
-	if in.NetworkMode != nil {
-		containerConfig.NetworkMode = *in.NetworkMode
+	if in.ContainerConfig.ConfigContainer != nil {
+		containerConfig.ConfigContainer = MapConfigContainer(in.ContainerConfig.ConfigContainer)
+	}
+
+	dagentConfig := in.ContainerConfig.Dagent
+
+	if dagentConfig != nil {
```

Comment:
You creating the `dagentConfig` and `craneConfig` variables even if there is no such config. Create variables only if we'll use those.